pkg/state/types.go

```diff
@@ -19,12 +19,28 @@ type EnvironmentTemplateData struct {
 	Values map[string]interface{}
 }
 
-// releaseTemplateData provides variables accessible while executing golang text/template expressions in releases of a helmfile YAML file
+// releaseTemplateData provides variables accessible while executing golang text/template expressions in release templates
+// and release values templates within a Helmfile YAML file.
 type releaseTemplateData struct {
```

Comment:
@joshes I realized that we already have a struct type for the template data used for rendering anything related to each release! So I refactored your change without modifying EnvironmentTemplateData.

Along the way I also realized that we already and unexpectedly exposed ReleaseSpec to the whole template in some case.

Fortunately, Helmfile had unit tests([`TestHelmState_executeTemplates`](https://github.com/roboll/helmfile/blob/3f02b866401ba7ef34c4c985de13151f3e51fb7b/pkg/state/state_exec_tmpl_test.go#L25)) that helped me to figure out which ReleaseSpec fields are intended to be exposed to the template. I extracted the fields to releseTemplateDataRelease(L34-L46).